dkg/dkg.go

```diff
@@ -136,24 +148,128 @@ func Run(ctx context.Context, conf Config) error {
 		return err
 	}
 
+	pubkeyToVerifiers := make(map[core.PubKey]*sharing.FeldmanVerifier)
+	pubkeyToPubshares := make(map[core.PubKey]map[uint32]*bls_sig.PublicKey)
+	for _, sh := range shares {
+		pk, err := tblsconv.KeyToCore(sh.PubKey)
+		if err != nil {
+			return err
+		}
+		pubkeyToVerifiers[pk] = sh.Verifier
+		pubkeyToPubshares[pk] = sh.PublicShares
+	}
+
+	// Signing, Aggregating and Writing Lock File
 	lock := cluster.Lock{
 		Definition: def,
 		Validators: dvs,
 	}
 
-	if conf.TestSigning {
-		sig, err := aggSignLockHash(ctx, tcpNode, nodeIdx, nil, lock, shares)
+	sigLockHash, err := signLockHash(lock, nodeIdx.ShareIdx, shares)
```

Comment:
split lockhash signing and depositdata signing into separate functions, there is too much going on this root function